rollouts/controllers/rollout_controller.go

```diff
@@ -115,15 +128,28 @@ func (r *RolloutReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 		return ctrl.Result{}, nil
 	}
 
-	err := r.reconcileRollout(ctx, &rollout)
+	packageDiscoveryClient := func() *packagediscovery.PackageDiscovery {
```

Comment:
nit: I think we can make this function a helper method on the `reconciler`. I think that will be easier for the code reader.

```Go
packageDiscoveryClient := r.GetPackageDiscoveryClient()
```

Reply:
Done.